### PR TITLE
Kubernetes PATCH operation

### PR DESCRIPTION
**Pull request by ediskandarov-cur**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
kubernetes

##### ANSIBLE VERSION
```
ansible 2.2.0.0
  config file = /Users/e.iskandarov/coinsph/infrastructure/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
Removes hard code in `k8s_update_resource` routine.

This PR adds ability to specify desired behavior on Kubernetes resource update.

### CODE REVIEW

**Comment by ediskandarov-cur** [comment]
@erjohnso please consider reviewing this PR

**Comment by erjohnso** [comment]
shipit

Looks good to me, thanks @toidi and sorry for taking so long!

**Comment by ryansb** [comment]
rebuild_merge

**Comment by ediskandarov-cur** [comment]
thanks, I'll fix that

File: lib/ansible/modules/clustering/kubernetes.py [hunk 1]
```diff
@@ -52,11 +52,13 @@
     required: false
     default: null
   patch_operation:
-    description:
-      - Specify patch operation for Kubernetes resource update. For details, see the description of PATCH operations at U(https://github.com/kubernetes/kubernetes/blob/release-1.5/docs/devel/api-conventions.md#patch-operations).
+    description: >
+      - Specify patch operation for Kubernetes resource update. For details, see the description of PATCH operations at
```
Comment (ediskandarov-cur) [inline lib/ansible/modules/clustering/kubernetes.py]: should'n it be like
```
    description:
      - >
        Specify patch operation for Kubernetes resource update. For details, see the description of PATCH operations at
        U(https://github.com/kubernetes/kubernetes/blob/release-1.5/docs/devel/api-conventions.md#patch-operations).
```

Comment (samdoran) [reply lib/ansible/modules/clustering/kubernetes.py]: @toidi That is the correct [folded scalar](http://www.yaml.org/spec/1.2/spec.html#style/block/folded) syntax.

**Pull request merged**